worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -69,18 +70,28 @@ function routeTask(rootDocument) {
                 var actionDetails = {
                     queueName: action.queueName,
                     scripts: action.scripts,
-                    customData: evalCustomData(arguments, action.customData),
-                    terminateOnFailure: action.terminateOnFailure
+                    customData: evalCustomData(arguments, action.customData)
                 };
 
                 rootDocument.getField('CAF_WORKFLOW_ACTION').add(action.name);
-                applyActionDetails(rootDocument, actionDetails);
+                applyActionDetails(rootDocument, actionDetails, terminateOnFailure);
                 break;
             }
         }
     }
 }
 
+function checkTerminateOnFailure(previousAction)
+{
+    if (previousAction) {
+        var previousIndex = ACTIONS.map(function (x) {
+            return x.name;
```

Comment:
Very minor, avoid x, at a minimum the first letter of the type is better. In this case a.